domain/machine/state/types.go

```diff
@@ -65,12 +68,30 @@ type machineLife struct {
 	LifeID life.Life `db:"life_id"`
 }
 
-// instanceID represents the struct to be used for the insatnce_id column within
+// instanceID represents the struct to be used for the instance_id column within
 // the sqlair statements in the machine domain.
 type instanceID struct {
 	ID string `db:"instance_id"`
 }
 
+// machineInstanceStatus represents the struct to be used for the status columns
+// of the machine_status and the machine_cloud_instance_status tables) within
+// the sqlair statements in the machine domain.
+type machineInstanceStatus struct {
```

Comment:
You could combine these types as we did for external controllers and addresses, get them in one query, then renormalise them.

This is a relatively frequent retrieval, so it's worth doing. Happy to have it done in a follow-up.

Reply:
Good idea! Done 👍